processing/data_collection/gazette/spiders/sp_santo_andre.py

```diff
@@ -0,0 +1,187 @@
+        }
+    }
+
+    allowed_domains = ["santoandre.sp.gov.br"]
+    start_urls = [
+        "http://www.santoandre.sp.gov.br/publicacao/edicao/consultaedicao.aspx"
+    ]
+    start_date = date(2010, 1, 1)
+    TODAY = datetime.now().strftime("%d/%m/%Y")
+    JAVASCRIPT_POSTBACK_REGEX = r"javascript:__doPostBack\('(.*)',''\)"
+    CURRENT_PAGE = 1
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield Request(url, callback=self.search_by_date)
+
+    def _get_form_params(self, response):
+        view_state = response.css("#__VIEWSTATE::attr(value)").get()
+        event_validation = response.css("#__EVENTVALIDATION::attr(value)").get()
+        return view_state, event_validation
+
+    def search_by_date(self, response):
+        VIEW_STATE, EVENT_VALIDATION = self._get_form_params(response)
+        yield FormRequest.from_response(
+            response,
+            callback=self.parse,
+            formname="aspnetForm",
+            formdata={
+                "__EVENTTARGET": "",
+                "__EVENTARGUMENT": "",
+                "__VIEWSTATE": VIEW_STATE,
+                "__EVENTVALIDATION": EVENT_VALIDATION,
+                "ctl00$ContentPlaceHolder1$dt_inicio": self.start_date.strftime(
+                    "%d/%m/%Y"
+                ),
+                "ctl00$ContentPlaceHolder1$dt_final": self.TODAY,
+            },
+            clickdata={"name": "ctl00$ContentPlaceHolder1$PsaToolBar1$btnSelecionar"},
+            method="POST",
+        )
+
+    def _save_pdf(self, response, path: Path, item):
+        if not isinstance(path, Path):
+            raise TypeError("O parâmetro path precisa ser uma instância de Path.")
+        path.parent.mkdir(parents=True, exist_ok=True)
+        with open(path, "wb") as f:
+            f.write(response.body)
+        yield item
+
+    def _parse_table_elements(self, element: Selector):
+        if not isinstance(element, Selector):
+            raise TypeError("O parâmetro element deve ser uma instancia de Selector.")
+        doPostBack, num_edicao, date = element.xpath(".//td").getall()
+        event_target = unquote(
+            re.search(self.JAVASCRIPT_POSTBACK_REGEX, doPostBack).groups()[0]
+        )
+        num_edicao = remove_tags(num_edicao)
+        date = dateparser.parse(remove_tags(date), languages=["pt"]).date()
+
+        return event_target, num_edicao, date
+
+    def _format_filename(self, num_edicao: str, event_target: str, date):
+        filename = sha1(
+            str.encode(
+                f"{self.allowed_domains[0].replace('.','')}-{event_target}-{num_edicao}-{date}"
+            )
+        ).hexdigest()
+        filename = f"{filename}.pdf"
+        file_path = Path(f"{FILES_STORE}full/{filename}")
+        return file_path
+
+    def _handle_navigation(self, response, current_page: int) -> Optional[str]:
+        """Responsável por realizar a identificação da próxima página a ser navegada na paginação.
```

Comment:
**issue**: Source code language must be English.

Reply:
that's fine... it's very common for me mix languages while i'm coding.